### ISSUE TITLE
Replacements within a hyperlink don't work

### ISSUE THREAD

**Issue by dtcooper** [six-two/mkdocs-placeholder-plugin#1]
I notice links aren't getting swapped out properly. Here are the steps to reproduce,

1. Provide a placeholder `LINK: https://example.com/`, and provide markdown `[xLINKx](xLINKx)`

2. Expect to see,
  ```html
  <a href="https://example.com/">
      https://example.com/
  </a>
  ```

3. Instead observe the `href="..."` inside the `<a>`/anchor tag does **not** get replaced, so we get,

```html
<a href="xLINKx">
   https://example.com/
</a>
```

This is observable and reproduced in the docs under "[Tests: General](https://mkdocs-placeholder-plugin.six-two.dev/tests/basic/)."

Thanks for the great plugin!

**Comment by six-two** [comment]
Hi David,

thank you very much for your comment.
The behavior described is kind of intentional, since I do not want to want the placeholder replacement to break the page by replacing text in inline-scripts/CSS/links. So I currently just replace visible content by filtering for it:
```javascript
const walker = document.createTreeWalker(root_element, NodeFilter.SHOW_TEXT);
...
while (node = walker.nextNode()) {
    replaced_str = node.nodeValue.replaceAll(search_regex, replacement_value);
...
```

I will look into the possibility of adding a global flag and/or adding an attribute to individual placeholders. However, this may introduce security vulnerabilities in your site, if user's set values such as `javascript:your_evil_code_here` for links.

BR,
six-two

**Comment by six-two** [comment]
I think that I fixed it in commit 109c275021173869507cc8b805e46886f3d74d3c. When you set the new [replace_everywhere](https://mkdocs-placeholder-plugin.six-two.dev/advanced-usage/#replace_everywhere) attribute to `true`, the placeholder will be replaced **everywhere** using a search and replace on `element.innerHTML`.

Is this the behavior you were expecting?

**Comment by dtcooper** [comment]
Exactly the behaviour I was looking for. Thank you!
